lib/server-middleware.js

```diff
@@ -16,20 +16,22 @@ module.exports = options => async (req, res, next) => {
   const optionSetter = setCustomValues(options, req)
   await Promise.all(customKeys.map(optionSetter))
 
-  const handler = new Handler({ req, res, next, options })
+  const handler = new Handler({
+    req, res, next, options
+  })
 
   // Refresh the token with the OAuth provider
   // useful for client side 401 handling
   if (handler.isRoute('refresh')) {
-    const { accessToken } = await handler.updateToken() || {}
-    const isAuthenticated = !!accessToken
-    res.writeHead(isAuthenticated ? 200 : 401, { 'Content-Type': 'application/json' })
+    const token = await handler.useRefreshToken()
 
-    if (isAuthenticated) {
-      const body = JSON.stringify({ accessToken })
+    if (token) {
+      res.writeHead(200, { 'Content-Type': 'application/json' })
+      const body = JSON.stringify(token)
```

Comment:
I think this will expose the refresh token to the client? We should only serialise the access token.

Reply:
I don't think the refresh token has that, but I will be extra safe and extract only the values we want `accessToken` & `expires`.